backend/scheduler/tasks.py

```diff
@@ -57,13 +59,27 @@ def execute_pipeline_task(
     name: Any,
 ) -> None:
     logger.info(f"Executing pipeline name: {name}")
-
     try:
         logger.info(f"Executing workflow id: {workflow_id}")
         tenant: Organization = (
             get_tenant_model().objects.filter(schema_name=org_schema).first()
         )
         with tenant_context(tenant):
+            subscription_plugin_available = (
+                os.environ.get("SUBSCRIPTION_PLUGIN_AVAILABLE", "False") == "True"
+            )
+            logger.info(f"Is subscription available: {subscription_plugin_available}")
+
+            if subscription_plugin_available and not etl_prerun_check(org_schema):
```

Comment:
Sugg: `not etl_prerun_check` Better to change to more meaning full name like validate_etl_run .

Reply:
Or we could be more explicit and just say `is_subscription_active()`